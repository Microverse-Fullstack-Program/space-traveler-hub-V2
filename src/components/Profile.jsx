import { useSelector } from 'react-redux';
import Card from 'react-bootstrap/Card';
import ListGroup from 'react-bootstrap/ListGroup';

function Profile() {
  const JoinedMissions = useSelector((state) => {
    const { missions } = state.missions;
    return missions.filter((mission) => mission.reserved);
  });

  const rockets = useSelector((state) => state.rockets.rockets);

  const reservedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div className="profile-wrapper">
      <div style={{ width: '100%' }}>
        <h4>My Missions</h4>
        <Card style={{ width: '100%' }}>
          {JoinedMissions.length ? JoinedMissions.map((mission) => (
            <ListGroup key={mission.id} variant="flush">
              <ListGroup.Item>{mission.mission_name}</ListGroup.Item>
            </ListGroup>
          ))
            : (
              <ListGroup>
                <ListGroup.Item>No Missions Joined</ListGroup.Item>
              </ListGroup>
            )}
        </Card>
      </div>
      <div style={{ width: '100%' }}>
        <h4>My Rockets</h4>
        <Card style={{ width: '100%' }}>
          {reservedRockets.length ? reservedRockets.map((rocket) => (
            <ListGroup key={rocket.id} variant="flush">
              <ListGroup.Item>{rocket.name}</ListGroup.Item>
            </ListGroup>
          ))
            : (
              <ListGroup>
                <ListGroup.Item>You have no reserved rockets</ListGroup.Item>
              </ListGroup>
            )}
        </Card>
      </div>
    </div>
  );
}

export default Profile;
