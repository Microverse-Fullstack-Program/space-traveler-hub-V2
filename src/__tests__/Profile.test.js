import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import setupStore from '../redux/store';
import Profile from '../components/Profile';

describe('Profile snapshot', () => {
  test('should render Profile page correctly', () => {
    render(
      <Provider store={setupStore({})}>
        <Profile />
      </Provider>,
    );
    const profile = document.querySelector('.profile-wrapper');
    expect(profile).toMatchSnapshot();
  });
});
