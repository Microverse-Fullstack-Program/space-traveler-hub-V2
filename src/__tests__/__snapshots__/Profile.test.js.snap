// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile snapshot should render Profile page correctly 1`] = `
<div
  class="profile-wrapper"
>
  <div
    style="width: 100%;"
  >
    <h4>
      My Missions
    </h4>
    <div
      class="card"
      style="width: 100%;"
    >
      <div
        class="list-group"
      >
        <div
          class="list-group-item"
        >
          No Missions Joined
        </div>
      </div>
    </div>
  </div>
  <div
    style="width: 100%;"
  >
    <h4>
      My Rockets
    </h4>
    <div
      class="card"
      style="width: 100%;"
    >
      <div
        class="list-group"
      >
        <div
          class="list-group-item"
        >
          You have no reserved rockets
        </div>
      </div>
    </div>
  </div>
</div>
`;
